File: 9. Palindrome Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x / 10 == 0:
            return True
        if x < 0 or x % 10 == 0:
            return False
        temp = 0
        while x > 0:
            temp = 10*temp + x % 10
            if x == temp or x / 10 == temp:
                return True
            x /= 10
        return False

File: 9. Palindrome Number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (Math.floor(x / 10) === 0)
        return true;
    if (x < 0 || x % 10 === 0)
        return false;
    let temp = 0;
    while (x > 0)
    {
        temp = 10*temp + x % 10;
        if (temp == x || temp == Math.floor(x / 10))
            return true;
        x = Math.floor(x / 10);
    }
    return false;
};
